### DISCUSSION TITLE
Portal Federation Fails

### DISCUSSION THREAD

**Discussion by SaskaTune** [Esri/arcgis-powershell-dsc#524] [Q&A]
I'm attempting a 3-machine deployment with SSL.  When it gets to the step where it needs to federate the server with portal, the process hangs.

Looking through ArcGIS.psm1 I think I found the problem on lines 2061 and 2062. There it is setting $PortalPort to 7443 (instead of 443 since I am using a web adapter) and setting $PortalContext to "arcgis" instead of the context defined in my JSON file ("portal").  By editing arcgis.psm1 and hardcoding those values to 443 and "portal I was able to make the setup succeed.

Now what I am not sure of is whether this is a bug or if I have simply missed something in my JSON configuration file?

**Comment by cameronkroeker** [comment]
Hi @SaskaTune,

Do you have SSLCertificates.Target set to portal, and SSLCertificates.CNameFQDN set to the internal portal hostname or to a DNS name?

https://github.com/Esri/arcgis-powershell-dsc/blob/1348e09ae3b6a7d69a3530d3e3766111b8425397/Modules/ArcGIS/ArcGIS.psm1#L2080

The logic is if ConfigData.Portal.InternalLoadBalancer is defined in the json config then it will use that with the ConfigData.Portal.InternalLoadBalancerPort to send the federation request to. However, if this isn't defined and SSLCertificates.CNameFQDN is defined with SSLCertificates.Target set to portal then it will use this with 7443 to send the federation request. If neither of these are present then it will send the federation request to https://portalMachineName.domain.com:7443.

I recommend if you have  SSLCertificates.Target set to portal then use the portal fully qualified machine name of the portal node for SSLCertificates.CNameFQDN.

Thanks,
Cameron K.
